sdk/src/types/block/output/feature/metadata.rs

```diff
@@ -434,35 +471,78 @@ pub(crate) mod irc_30 {
 
 #[cfg(feature = "serde")]
 pub(crate) mod dto {
-    use alloc::borrow::Cow;
+    use alloc::{collections::BTreeMap, format};
 
-    use serde::{Deserialize, Serialize};
+    use serde::{de, Deserialize, Deserializer, Serialize};
+    use serde_json::Value;
 
     use super::*;
-    use crate::utils::serde::cow_boxed_slice_prefix_hex_bytes;
 
-    #[derive(Serialize, Deserialize)]
-    struct MetadataFeatureDto<'a> {
+    #[derive(Serialize)]
+    struct MetadataFeatureDto {
         #[serde(rename = "type")]
         kind: u8,
-        #[serde(with = "cow_boxed_slice_prefix_hex_bytes")]
-        data: Cow<'a, BoxedSlicePrefix<u8, MetadataFeatureLength>>,
+        entries: BTreeMap<String, String>,
+    }
+
+    impl<'de> Deserialize<'de> for MetadataFeature {
+        fn deserialize<D: Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
+            let value = Value::deserialize(d)?;
+            Ok(
+                match value
+                    .get("type")
+                    .and_then(Value::as_u64)
+                    .ok_or_else(|| de::Error::custom("invalid metadata type"))? as u8
```

Comment:
if the type is out of u8 bounds for some (malicious or accidental) reason then this method would still panic instead of returning an error and allowing to continue on, so I guess we need to do a bounds check before performing `as u8`  because there's no `Value::as_u8` (in serde_json) that would do it for us?

Reply:
`Value::as_u8` doesn't exist, but I used `u8::try_from` now :+1: